design-documents/graph-ql/coverage/search.md

```diff
@@ -0,0 +1,79 @@
+# Queries
+
+```graphql
+type Query {
+    multiSearch(phrase: String!, productSize: Int = 10): MultiSearchResponse!
+    #Filter supports multiple clauses which will be wrapped in logical AND operator
+    productSearch(phrase: String!, filter: [SearchClauseInput], sort: [ProductSearchSortInput]): ProductSearchResponse!
+}
+
+input ProductSearchSortInput
+{
+    attribute: String!
+    direction: SortEnum!
```

Comment:
why should we have direction as a mandatory attribute? 
Most of RDBMS just use a default value of ASC / DESC

Reply:
Here is official elastic search documentation:
```
The order defaults to desc when sorting on the _score, and defaults to asc when sorting on anything else.
```
I don't like this uncertainty

`ProductSearchSortInput` is optional so we can apply some default order, but if developer asks for custom ordering I think it's better to ask for direction as well.

Reply:
I like @maghamed's idea of having a default.

> I don't like this uncertainty

I don't see much uncertainty from that quote, if we're only talking about ES.

But, are we planning to only support Elastic Search? If we're ever going to swap search providers, I think it would make sense for the application to define the default so it's consistent.

> ProductSearchSortInput is optional so we can apply some default order

I think an application-defined default sort order would be great 👍

Reply:
Starting to agree with you more here @kokoc. I took a look again at the usage of `ProductSearchSortInput`, and I guess it will always be specified right next to `attribute_code`, so generic `to`/`from` `float` will probably cover that 👍

Reply:
I think there is no need for any changes. We have the optional `ProductSearchSortInput`, so we can apply default sort. However, the user needs to specify direction for every custom sort. Does it work?